src/lib/graveyard-container.js

```diff
@@ -0,0 +1,19 @@
+class Graveyard {
+  constructor(refObject) {
+    refObject.onInserted.add(this.onInserted);
+  }
+
+  onInserted(container, insertedObjects, player) {
+    for (let i = 0; i < insertedObjects.length; i++) {
+      const name = insertedObjects[i].getTemplateName()
```

Comment:
Take a look at lib/object-namespace.js for a bunch of isX and parseX functions.

Reply:
Nice, `object-namespace.js` will be very useful here! Do we have any constants yet defining known types of things yet? If not, how about adding them as properties of `ObjectNamespace`?
```js
class ObjectNamespace {
  ...
  OBJECT_TYPES = {
    'Card': 'card',
    'Token': 'token',
    'Tile': 'tile',
    'Unit': 'unit',
    'Bag': bag'
    ...
  }
  ...
  DECK_TYPES = {
    'Action': 'action',
    'Agenda': 'agenda',
    'Promissory': 'promissory',
    ...
  }
  ...
}

```

Reply:
Interesting.  Not at the moment, how would you use it?

ObjectNamespace could expose a method to give just the first item in the type.  There will probably be a small number of new types added but what's there now ought to be fairly complete.

For decks, if you call ObjectNamespace.parseCard the result will have a "deck" field with the "deck name" portion of the type.

Reply:
When I wrote that I was thinking of adding that exact method, to return the first item in the type, which I'd then switch through to decide how to return the given object home. But I've now coded that with `if(ObjectNamespace.isX)` instead, so I don't need those types in another file yet.

If we start needing those strings outside of `object-namespace.js` then defining them up-front to prevent hard-coded strings proliferating around the codebase would be a good idea. Explicitely Ddefining the recognised types all in a single place is also helpful so you can see the available options at a glance without having to dig through the code (thinking of people new to the project especially).

Reply:
That's fair.  My thinking is that if something needs to check a type, there should be a `ObjectNameSpace.isX` method so those are _sort of_ documenting it.  I do plan on writing up a wiki page with more details on the larger namespace design -- that might work for newcomers (as well as sanity checking later).

Reply:
Yeah, the `ObjectNamespace` class is definitely helping somewhat and a wiki would also. Personally I tend to prefer code that self-documents though, as wikis have an uncanny ability to get out-of-date **very** quickly 😅